nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch5.java

```diff
@@ -0,0 +1,451 @@
+                        kmf.init(keyStore, sslService.getKeyStorePassword().toCharArray());
+                        final TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory
+                                .getDefaultAlgorithm());
+                        tmf.init(keyStore);
+                        SSLContext context = SSLContext.getInstance(sslService.getSslAlgorithm());
+                        context.init(kmf.getKeyManagers(), tmf.getTrustManagers(), new SecureRandom());
+                        httpClientBuilder = httpClientBuilder.setSSLContext(context);
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+
+                if (username != null && password != null) {
+                    final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                    credentialsProvider.setCredentials(AuthScope.ANY,
+                            new UsernamePasswordCredentials(username, password));
+                    httpClientBuilder = httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                }
+
+                return httpClientBuilder;
+            }
+        })
+        .setRequestConfigCallback(requestConfigBuilder -> requestConfigBuilder.setConnectTimeout(connectTimeout)
+                .setSocketTimeout(readTimeout))
+        .setMaxRetryTimeoutMillis(retryTimeout);
+
+        this.client = builder.build();
+    }
+
+    private Response runQuery(String query, String index, String type) throws Exception {
+        StringBuilder sb = new StringBuilder()
+                .append("/")
+                .append(index);
+        if (type != null && !type.equals("")) {
+            sb.append("/")
+                .append(type);
+        }
+
+        sb.append("/_search");
+
+        HttpEntity queryEntity = new NStringEntity(query, ContentType.APPLICATION_JSON);
+
+        return client.performRequest("POST", sb.toString(), Collections.emptyMap(), queryEntity);
+    }
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        try {
+            if (this.client == null) {
+                setupClient(context);
+            }
+
+            String query = context.getProperty(QUERY).evaluateAttributeExpressions(flowFile).getValue();
+            String index = context.getProperty(INDEX).evaluateAttributeExpressions(flowFile).getValue();
+            String type = context.getProperty(TYPE).evaluateAttributeExpressions(flowFile).getValue();
+
+            Response resp = runQuery(query, index, type);
+
+            InputStream content = resp.getEntity().getContent();
+            String rawJson = IOUtils.toString(content, "UTF-8");
```

Comment:
Is the response guaranteed to be in UTF-8 format?

Reply:
Apparently Lucene requires it: https://stackoverflow.com/a/23044215